pkg/registry/extensions/rest/thirdparty_controller.go

```diff
@@ -76,19 +78,31 @@ func (t *ThirdPartyController) SyncResources() error {
 }
 
 func (t *ThirdPartyController) syncResourceList(list runtime.Object) error {
-	existing := sets.String{}
+	existing := map[metav1.GroupVersionResource]sets.Empty{}
 	switch list := list.(type) {
 	case *extensions.ThirdPartyResourceList:
 		// Loop across all schema objects for third party resources
 		for ix := range list.Items {
 			item := &list.Items[ix]
 			// extract the api group and resource kind from the schema
-			_, group, err := thirdpartyresourcedata.ExtractApiGroupAndKind(item)
+			kind, group, err := thirdpartyresourcedata.ExtractApiGroupAndKind(item)
```

Comment:
maybe we can return `GroupKind` here, seems we cannot extract `GroupResource` from item, wdyt?